src/main/java/data/TaskManager.java

```diff
@@ -473,7 +473,8 @@ private static LocalDate findDateFromDayNumber(WeekView weekView, MonthView mont
      * @param date The date of the task to be marked.
      */
     private void handleMarkingOfTask(int taskIndex, LocalDate date) {
-
+        logger.log(Level.INFO, "Marking task at index " + taskIndex + " for date " + date);
```

Comment:
Good job on adding logging for every major function that doesn't have it yet. This will be useful for recording the inputs and outputs, and all local user interactions!